### PR TITLE
RavenDB-15133 - Support GUID, DateTime, DateTimeOffset and enum constants in first class patch

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-15133

### CODE REVIEW

File: test/SlowTests/Issues/RavenDB-15133.cs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                Map = doc => from documents in doc
+                             select new Result
+                             {
+                                 Name = documents.Name
+                             };
+            }
+        }
+
+        [Fact]
+        public async Task CanSelectIdField()
```
Comment (ayende) [inline test/SlowTests/Issues/RavenDB-15133.cs]: How does this relates to the change you made?

File: src/Raven.Client/Util/JavascriptConversionExtensions.cs [hunk 2]
```diff
@@ -1875,6 +1876,47 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
...
+            private ConstantSupport()
+            {
+            }
+
+            public override void ConvertToJavascript(JavascriptConversionContext context)
+            {
+                if (!(context.Node is ConstantExpression ce))
+                    return;
+
+                var isDateTime = ce.Type == typeof(DateTime);
```
Comment (ayende) [inline src/Raven.Client/Util/JavascriptConversionExtensions.cs]: Would be better to do using a case here, no?

File: test/SlowTests/Issues/RavenDB-15133.cs (hunk 1 above)
Comment (grisha-kotler) [reply test/SlowTests/Issues/RavenDB-15133.cs]: forgot to push the test

File: src/Raven.Client/Util/JavascriptConversionExtensions.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Client/Util/JavascriptConversionExtensions.cs]: changed

**Review by ppekrol — APPROVED** [review]

**Review by ayende — APPROVED** [review]

**Pull request merged**